backend/services/rag_service.py

```
@@ -197,71 +524,324 @@ def start_index_job(self, document_ids: Optional[List[str]] = None) -> Dict[str,
+        
+        # Use our existing chunker logic but return the whole thing
+        try:
+            if suffix == ".pdf":
+                with open(file_path, "rb") as f:
+                    reader = PyPDF2.PdfReader(f)
+                    return "\n".join([page.extract_text() for page in reader.pages])
+            
+            if suffix == ".docx":
+                import docx
+                doc = docx.Document(file_path)
+                return "\n".join([para.text for page in doc.paragraphs])
```

Ruff (0.15.9)

[error] Undefined name `para`

(F821)